bitrise.yml

```diff
@@ -585,6 +597,50 @@ workflows:
                 exit 1
             fi
         is_always_run: true
+  Decision_to_run_UI_Tests:
+    steps:
+    - script@1:
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -e
+
+            echo "Commit on origin/main"
+            COMMIT_ON_MAIN_ORIGIN=`git log -n 1 origin/main --pretty=format:"%H"`
+            echo $COMMIT_ON_MAIN_ORIGIN
+
+            echo "Files changed with origin"
+            echo `git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_ORIGIN`
+
+            echo "Commit on main branch"
+            echo $GIT_CLONE_COMMIT_HASH
+            COMMIT_ON_MAIN_PR=`git log -n 2 main --pretty=format:"%H" | tail -n 1`
+            echo $COMMIT_ON_MAIN_PR
+
+            filesChanged=""
+            if [[ $COMMIT_ON_MAIN_PR == $COMMIT_ON_MAIN_ORIGIN ]]
+            then
+                echo "First commit, use origin/main"
+                filesChanged=`git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_ORIGIN`
+                echo $filesChanged
+            else
+                echo "Commits on PR"
+                echo "Files changed betwee commits"
+                echo `git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_PR`
+                filesChanged=`git diff --name-only $GIT_CLONE_COMMIT_HASH $COMMIT_ON_MAIN_PR`
+                echo $filesChanged
+            fi
+
+            doNotRunTests='^(\.[a-zA-Z]+\.y*|Docs/*|Sync*/*|.github/|[a-zA-Z]+\.md|[a-zA-Z]+\.sh|[a-zA-Z0-9]+[-_]+[a-zA-Z]+\.sh)|fastlane/*|taskcluster/*|L10nSnapshotTests/*|Shared/[a-zA-Z]+\.lproj/*|Shared/[A-Za-z]/*|WidgetKit/[a-zA-Z]+\.lproj/*|Client/*/[a-zA-Z]+\.lproj/*'
```

Comment:
Just double checking if you need to escape the `/` characters `\/`, tested the entire query on https://regexr.com with `Docs/` and `Shared/` and it errored.

Will this run if a commit contains both code where we want UI tests to run and code where we don't need UI tests to run? E.g, I commit a change to `Docs` as well as core browser code.

Reply:
mm locally it works for me without scaping those, but I will double check to be sure it also works on CI.
Yes, I have added the logic to cover these scenarios:
No changes: UI tests do not run

> Changes in no "app" files AND "app" files: UI tests run
> Changes in "app" files: UI tests run
> Changes in no "app" files: UI tests do not run